en/developer/frameworks/articles/info-framework/05-creating-information-item-selector.markdown

```diff
@@ -0,0 +1,27 @@
+---
+header-id: custom-selecting-of-information-with-infoitemselector
+---
+
+# Selecting Custom information Types with `InfoItemSelector`
+
+[TOC levels=1-4]
+
+The [Content fragment](/docs/7-2/user/-/knowledge_base/u/content-page-management-interface#section-builder) 
+and [itemSelector fragment configuration](/docs/7-2/reference/-/knowledge_base/r/fragment-configuration-types#item-selector-configuration) 
+let you display a single Web Content, Blog Entry, or document in a fragment. To 
+make your custom content type available for selection, follow these steps:
+
+1. Create an [`InfoItemRenderer`](/docs/7-2/frameworks/-/knowledge_base/f/custom-rendering-of-information-with-infoitemrenderer) to render the content.
+
+2.  Create an [Item Selector](/docs/7-2/frameworks/-/knowledge_base/f/selecting-entities-with-an-item-selector) 
+    to select the content that has a `InfoItemItemSelectorReturnType`.
```

Comment:
This is slightly different, you have to create an ItemSelectorView class that implements both InfoItemSelectorView and ItemSelectorView<InfoItemItemSelectorCriterion>
and as you mention in the ItemSelectorReturnType use InfoItemItemSelectorReturnType

I think we should link to an example like [this](https://github.com/liferay/liferay-portal/blob/df3fe0e6bca1341a8d1333f9f8d423a512fa49d3/modules/apps/blogs/blogs-web/src/main/java/com/liferay/blogs/web/internal/item/selector/BlogsEntryItemSelectorView.java) 

What do you think?

Reply:
That's internal, so we can't link to it, but I can show example code that is similar. Also, if we're providing example code, we want to show it in the article and explain it. We don't want to link to code and tell developers to go look at it without context. Thanks for the feedback! I'll make updates based on it and share the changes.

Reply:
What do you understand by internal? This is public on github and it is a working example. We don't usually link the the source code in the docs?

Reply:
Sorry, you're right :-) it is a public repo. 

We don't usually link to source code unless it's a class that the developer needs to build the project. If there is an example, we prefer to show/explain it in the docs. All I mean is that we don't want to point a developer to source code by itself as an example and tell them to go figure it out themselves. We want to explain it as well in the docs.